### DISCUSSION TITLE
.NET 5.0 - Docker Repo Name Change

### DISCUSSION THREAD

**Discussion by mthalman** [dotnet/dotnet-docker#3667] [Announcements]
# .NET 5.0 - Docker Repo Name Change

With the release of .NET 5.0 Preview 4, the Docker repositories where 5.0 images are located have been changed to no longer include `core` in the repo name.

## Details

In order to better align with the .NET 5.0 branding, "core" is being dropped from the repository name. See https://github.com/dotnet/dotnet-docker/issues/1765 for details.

Previous Repositories:

* [dotnet/core](https://hub.docker.com/_/microsoft-dotnet-core): .NET Core
* [dotnet/core/sdk](https://hub.docker.com/_/microsoft-dotnet-core-sdk/): .NET Core SDK
* [dotnet/core/aspnet](https://hub.docker.com/_/microsoft-dotnet-core-aspnet/): ASP.NET Core Runtime
* [dotnet/core/runtime](https://hub.docker.com/_/microsoft-dotnet-core-runtime/): .NET Core Runtime
* [dotnet/core/runtime-deps](https://hub.docker.com/_/microsoft-dotnet-core-runtime-deps/): .NET Core Runtime Dependencies
* [dotnet/core/samples](https://hub.docker.com/_/microsoft-dotnet-core-samples/): .NET Core Samples
* [dotnet/core-nightly](https://hub.docker.com/_/microsoft-dotnet-core-nightly/): .NET Core (Preview)
* [dotnet/core-nightly/sdk](https://hub.docker.com/_/microsoft-dotnet-core-nightly-sdk): .NET Core SDK (Preview)
* [dotnet/core-nightly/aspnet](https://hub.docker.com/_/microsoft-dotnet-core-nightly-aspnet): ASP.NET Core Runtime (Preview)
* [dotnet/core-nightly/runtime](https://hub.docker.com/_/microsoft-dotnet-core-nightly-runtime): .NET Core Runtime (Preview)
* [dotnet/core-nightly/runtime-deps](https://hub.docker.com/_/microsoft-dotnet-core-nightly-runtime-deps): .NET Core Runtime Dependencies (Preview)

New repositories for .NET 5 going forward:

* [dotnet](https://hub.docker.com/_/microsoft-dotnet)
* [dotnet/sdk](https://hub.docker.com/_/microsoft-dotnet-sdk)
* [dotnet/aspnet](https://hub.docker.com/_/microsoft-dotnet-aspnet)
* [dotnet/runtime](https://hub.docker.com/_/microsoft-dotnet-runtime)
* [dotnet/runtime-deps](https://hub.docker.com/_/microsoft-dotnet-runtime-deps)
* dotnet/samples -> available once .NET 5.0 releases as GA
* [dotnet-nightly](https://hub.docker.com/_/microsoft-dotnet-nightly)
* [dotnet-nightly/sdk](https://hub.docker.com/_/microsoft-dotnet-nightly-sdk)
* [dotnet-nightly/aspnet](https://hub.docker.com/_/microsoft-dotnet-nightly-aspnet)
* [dotnet-nightly/runtime](https://hub.docker.com/_/microsoft-dotnet-nightly-runtime)
* [dotnet-nightly/runtime-deps](https://hub.docker.com/_/microsoft-dotnet-nightly-runtime-deps)

.NET 5.0 Preview 4 will be used as a transition period for this repo name change. This means that the 5.0 floating tags in the old repo will be updated to reference the Preview 4 images. For example, the .NET 5.0 runtime image will be tagged with both `mcr.microsoft.com/dotnet/core/runtime:5.0` and `mcr.microsoft.com/dotnet/runtime:5.0`. This tagging strategy will end once Preview 5 is released. At that point, all 5.0 tags using the old repo name will no longer be updated. In addition, all 5.0 floating tags will be removed from the old repo for the GA release of 5.0 in order to avoid accidental usage.

**Comment by tristan-hyams** [comment]
@mthalman Hey Matt, new templates are still pointing to `/core/` directories for Preview 6. People are going to hit a weird wall telling them (rightly so) that their build version of Buster-Slim is not found.

Current:  
```
FROM mcr.microsoft.com/dotnet/core/aspnet:5.0-buster-slim AS base  
...
FROM mcr.microsoft.com/dotnet/core/sdk:5.0-buster AS build
```

Needs to change to:  
```
FROM mcr.microsoft.com/dotnet/aspnet:5.0-buster-slim AS base  
...
FROM mcr.microsoft.com/dotnet/sdk:5.0-buster AS build
```

Base dockerfiles for AspNetCore templates (maybe others) are still going to the legacy path despite the docker images not actually being available for Preview6 under the `/core/` routing.

I recommend pushing the build of Preview6 also to the old route, then more publicly make them aware that going forward, RC1, or Preview7 etc is under the new route WHILE also making sure fresh Dockerfile files in templates are correct.

Thanks for all your hard work, regardless!

**Comment by mthalman** [comment]
Thanks, I'll follow up with the VS team to see what's going on here since this was supposed to have been updated.

**Comment by mthalman** [comment]
What version of VS are you running (Help -> About Microsoft Visual Studio)?  You should be running the latest update of 16.6 to ensure you have the changes that fix this.

**Comment by tristan-hyams** [comment]
Wasn't sure which frameworks you wanted to see but here is the full thing.

```
Microsoft Visual Studio Professional 2019
Version 16.6.3
VisualStudio.16.Release/16.6.3+30225.117
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Professional

Visual C++ 2019   00435-60000-00000-AA986
Microsoft Visual C++ 2019

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools 2019   16.6.954.58314
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2020603+b3ac44798c16fff5b95dbcfe62dea84aa9a1bd72
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.6.954.58314
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.6.954.58314
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.4.6000.1
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   16.6.954.58314
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.4.6000.1
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   3.6.0-4.20251.5+910223b64f108fcf039012e0849befb46ace6e66
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.0 (d16-2@8b56e20)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

GitHub.VisualStudio   2.11.104.25275
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure HDInsight Azure Node   2.4.6000.1
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.4.6000.1
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   16.0
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.4.6000.1
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.30212.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.50+g25aae5a24a.R
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.6.17 (9692114)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Office Developer Tools for Visual Studio   16.0.30007.00
Microsoft Office Developer Tools for Visual Studio

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62006.03190
Microsoft SQL Server Data Tools

TermWindow Extension   1.0
TermWindow Visual Studio Extension Detailed Info

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   16.0.20417.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.6.0-4.20251.5+910223b64f108fcf039012e0849befb46ace6e66
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.9.1.0 for F# 4.7   16.6.0-beta.20217.4+1c969cac25e2d38d71872efe6c8226029e42bb59
Microsoft Visual F# Tools 10.9.1.0 for F# 4.7

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions (Preview)   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Kubernetes   1.0
Visual Studio Tools for Kubernetes

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Workflow Manager Tools 1.0   1.0
This package contains the necessary Visual Studio integration components for Workflow Manager.

Xamarin   16.6.000.1062 (d16-6@41e1f0e)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.6.0.318 (remotes/origin/d16-6@66afc5421)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.6.40 (1f1466f)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   10.3.1.4 (d16-6/3a10de9)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: 165f4b0
    Java.Interop: xamarin/java.interop/d16-6@2cab35c
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.31.1@49232bc
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-6@bfb66f3


Xamarin.iOS and Xamarin.Mac SDK   13.18.2.1 (29c4ea7)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

**Comment by mthalman** [comment]
Ok, thanks.  I've confirmed that this is an issue with 16.6.

**Comment by mthalman** [comment]
The plan is to "backport" the latest 5.0 images to the original repo/tag names: https://github.com/dotnet/dotnet-docker/pull/2087

**Comment by mthalman** [comment]
@houseofcat - This should now be fixed.

**Comment by NicolasReyDotNet** [comment]
Hi, @mthalman  I did upgrade from 3.1 to 5 very smoothly, thank you.

I noticed that there is an "equivalent" to previous 3.1 buster slim version for both aspnet and sdk : 
`mcr.microsoft.com/dotnet/core/aspnet:3.1-buster-slim` => `mcr.microsoft.com/dotnet/aspnet:5.0-buster-slim`
and
`mcr.microsoft.com/dotnet/core/sdk:3.1-buster-slim` => `mcr.microsoft.com/dotnet/sdk:5.0-buster-slim`

But is there any "equivalent" to previous `mcr.microsoft.com/dotnet/core/sdk:3.1-buster` ?
Is it `mcr.microsoft.com/dotnet/sdk:5.0` ?

Many thanks for your help

**Comment by mthalman** [comment]
> But is there any "equivalent" to previous `mcr.microsoft.com/dotnet/core/sdk:3.1-buster` ?
> Is it `mcr.microsoft.com/dotnet/sdk:5.0` ?

@NicolasReyDotNet - The equivalent to `mcr.microsoft.com/dotnet/core/sdk:3.1-buster` is `mcr.microsoft.com/dotnet/sdk:5.0-buster-slim`.  Because of https://github.com/dotnet/dotnet-docker/issues/2376, the 5.0 SDK for Buster is now based on `aspnet:5.0-buster-slim` instead of `buildpack-deps:buster-scm`.  There didn't exist a slim variant of `buster-scm` so it was the full Buster in 3.1.  But with 5.0, it's using the slim variant.

**Comment by NicolasReyDotNet** [comment]
@mthalman 
Got it thank you, and congratz for your amazing job! 👍
